std/algebra/emulated/sw_bn254/pairing.go

```diff
@@ -232,6 +232,69 @@ func (pr Pairing) AssertIsEqual(x, y *GTEl) {
 	pr.Ext12.AssertIsEqual(x, y)
 }
 
+func (pr Pairing) AssertIsOnCurve(P *G1Affine) {
```

Comment:
I guess this is the same check as in `sw_emulated.Curve.AssertIsOnCurve`? If so, then we can instead wrap that method to have a single point of implementation.

This needs redefinind `Pairing` struct though. I can do it.

Reply:
Yup it's the same thing.